index.js

```diff
@@ -1,20 +1,257 @@
+  },
+];
+
+const openpop = document.querySelectorAll('.button1');
+const secpop = document.querySelector('.Secpopupwin');
+const closepop = document.querySelector('.Closepop');
+const popupimage = document.querySelector('.popupWindowImage');
+const popuppara = document.querySelector('.headingpop');
+const popupServer = document.querySelector('#serverWin');
+const popupLorem = document.querySelector('.Lorem');
+const live = document.querySelector('#live');
+const source = document.querySelector('#source');
+
+const openpopwin = () => {
+  secpop.classList.add('active');
+};
+
+const closepopwin = () => {
+  secpop.classList.remove('active');
+};
+
+closepop.addEventListener('click', closepopwin);
+
+openpop.forEach((windowpop) => {
+  windowpop.addEventListener('click', () => {
+    openpopwin();
+
+    projectPopupWindow.forEach((windowpopup) => {
+      if (windowpopup.id === windowpop.id) {
+        popupimage.src = windowpopup.image;
+        popuppara.textContent = windowpopup.heading;
+        popupServer.textContent = '';
+        windowpopup.technologies.forEach((tech) => {
+          const techlist = document.createElement('li');
+          techlist.innerText = tech;
+          popupServer.appendChild(techlist);
+        });
+        popupLorem.innerHTML = windowpopup.paragraph;
+        live.href = windowpopup.button1;
+        source.href = windowpopup.button2;
+      }
+    });
+  });
+});
+
+const formValidation = document.querySelector('form');
+const error = document.querySelector('.display-error');
+
+const isUpperCaseEmail = (email) => {
+  if (/[A-Z]/.test(email)) {
+    return true;
+  }
+  return false;
+};
+
+formValidation.addEventListener('submit', (event) => {
+  event.preventDefault();
+  const email = event.target.email.value;
+  const validEmail = !isUpperCaseEmail(email);
+
+  if (validEmail) {
+    error.innerHTML = '';
+    formValidation.submit();
+  } else {
+    error.style.color = 'red';
+    error.innerHTML = 'Please Email should be in lowercase';
+  }
+});
+
+// Retrieve the form element and input element
+const form = document.getElementById('myForm');
+const input = document.getElementById('fname');
+const email = document.getElementById('email');
+const textArea = document.getElementById('type');
+
+// Add an event listener to the form submission
+form.addEventListener('submit', (e) => {
+  e.preventDefault(); // Prevent form submission
+
+  // Store the input value in localStorage
+  localStorage.setItem('myInputValue', input.value);
+  localStorage.setItem('hell', email.value);
+  localStorage.setItem('heaven', textArea.value);
+  // Reset the form
+  form.reset();
+});
+
+// Check if the input value is stored in localStorage
+if (localStorage.getItem('myInputValue')) {
+  // Set the input value from localStorage
+  input.value = localStorage.getItem('myInputValue');
+  email.value = localStorage.getItem('hell');
+  textArea.value = localStorage.getItem('heaven');
+}
```

Comment:
- [ ] While you did a fantastic job storing the form using local storage, as a project requirement this should be stored in a single object as you can see below :+1:.

```js
const formData = {
    name: nameInput.value,
    email: emailInput.value,
    body: bodyInput.value,
};
```

- [ ] Also, could you make sure that you are using the correct event to save the information because now I noticed that you are calling the function when the user submits the form which is giving you some errors. I think using input event listener to deal with most of the problems would be better.
 

https://github.com/Mike47ip/mobilePortfolioWeb/assets/105117832/fecffb7b-8f35-4020-9481-e65b102cc9cb